system-helper/org.freedesktop.Flatpak.rules.in

```diff
@@ -14,7 +14,11 @@ polkit.addRule(function(action, subject) {
 
 polkit.addRule(function(action, subject) {
     if (action.id == "org.freedesktop.Flatpak.override-parental-controls") {
+        if (subject.isInGroup("@privileged_group@")) {
+            return polkit.Result.YES;
+        } else {
             return polkit.Result.AUTH_ADMIN;
+        }
```

Comment:
I’m hesitant about this change. `override-parental-controls` is only checked if the app being installed exceeds the user’s parental controls policy (even if the user is an admin). Returning `AUTH_ADMIN` should require the user to authenticate as an administrator — if the user *is* an administrator, that just means entering their password; if the user *isn’t* an administrator, that means finding an administrator and persuading them to enter their password.

Returning `YES` if the user is an administrator means they won’t be asked to confirm, which essentially unconditionally disables parental controls for all admin users.

Reply:
Yes, I agree, the problem is that currently the default parental controls settings for admins have system installation disabled by default, so the override action is always checked on check_parental_controls for admins (can be seem by running `malcontent-client get` from an admin user).

Maybe we should change the default parental controls settings for admin users to have system installation enabled by default?

Reply:
Also worth noting that the other actions are already allowed for admin users by the current polkit policy which return `YES` for `app-install` and co if the user is an admin.

Reply:
Fwiw, we also did ignore parental controls checks for admins on `eos3.6` and I think the idea is that admins can always install/update/... apps (we dont even enable the options to disable install in the parental controls settings UI for admins) - https://github.com/endlessm/flatpak/blob/eos3.6/system-helper/org.freedesktop.Flatpak.rules.in#L11.

The checks should still be done when running an app though, so admins can still block running apps if I understood correctly.

Reply:
> Maybe we should change the default parental controls settings for admin users to have system installation enabled by default?

I think that would be preferable.

> Also worth noting that the other actions are already allowed for admin users by the current polkit policy which return `YES` for `app-install` and co if the user is an admin.

It doesn’t matter what else the admin is allowed to do. The idea was that admins would be prompted before installing apps which are above their parental controls policy level — see https://phabricator.endlessm.com/T26893.

In this situation, I don’t think it especially matters what the behaviour was on `eos3.6`. What matters are the ACs for https://phabricator.endlessm.com/T26893.

> The checks should still be done when running an app though, so admins can still block running apps if I understood correctly.

The checks for running an app and installing an app are very different (regardless of whether you’re an admin or a normal user). When you install an app, its OARS ratings are compared to your user’s OARS level. When you run an app, its app ID (or path) is compared to a blacklist. `override-parental-controls` is only consulted when installing apps (see `check_parental_controls()` in `flatpak-run.c` — it doesn’t check polkit).

Perhaps we should allow an override when running apps, but that’s an issue for a separate task.

Reply:
My last 2 cents before vanishing for my days off :).

Maybe we could do something like:
- always allow `app-install` for all users via polkit rules as you mentioned  [here](https://phabricator.endlessm.com/T27854#757957)
- always run `override-parental-controls` on `check_parental_controls` passing hints for whether system app installation is allowed and the rating is appropriate
- on polkit rules for `override-parental-controls` do something like (pseudo code):
```
# dont show polkit challenge if system app install is allowed and rating is appropriate
# for admins we always consider system app install is allowed
if (user.admin==false and system_app_install_allowed and rating_is_appropriate) or
   (user.admin==true and rating_is_appropriate):
    return YES

if user.admin==false and not system_app_install_allowed:
    return AUTH_ADMIN

if not rating_is_appropriate:
    return AUTH_ADMIN

return NOT_HANDLED
```

The above doesnt require changing the default setting for system app install for admin users as it always consider it to be enabled - but we could still change that and always check if system app install is allowed. It would also always (admin or not) run a challenge if the rating is not appropriate.

If we go with that we should probably rename `override-parental-controls` with `validate-parental-controls` or something along this line.

Reply:
I’m experimenting with something along those lines, yes, thanks for the clarity of thinking.

This work is going to continue well into tomorrow though.